src/text.rs

```diff
@@ -596,7 +599,48 @@ fn shape_run(
                     if let Some(max_width) = max_width {
                         if result.width + word.width >= max_width {
                             word_break_reached = true;
-                            break;
+                            if first_word_in_paragraph {
+                                // search for the largest prefix of the word that can fit
+                                let mut bytes_included = 0;
+                                let mut subword_width = 0.0;
+                                let target_width = max_width - result.width;
+                                for glyph in word.glyphs.iter() {
+                                    bytes_included = glyph.byte_index;
+                                    let glyph_width = glyph.advance_x + paint.letter_spacing;
+
+                                    // nuance: we want to include the first glyph even if it breaks
+                                    // the bounds. this is to allow pathologically small bounds to
+                                    // at least complete rendering
+                                    if subword_width + glyph_width > target_width && bytes_included != 0 {
```

Comment:
I wonder if this should be `>=`?

Reply:
I don't think it matters, because we're comparing floats, but I've fixed it!

Reply:
Yeah, it's more of a matter of consistency :). Thanks!